### PR TITLE
added entity-relationships and repositories

### PR DESCRIPTION
**Pull request by DashDeipayan**
- https://github.com/Real-Dev-Squad/skill-tree-backend/issues/15

### CODE REVIEW

File: skill-tree /src/main/java/com/RDS/skilltree/Endorsement/EndorsementModel.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    private SkillModel skill;
+
+    @Column(name = "endorsement_status")
+    private EndorsementStatus status;
+
+    @Column(name = "created_at", nullable = false)
+    @GeneratedValue
+    private Instant createdAt;
+
+    public EndorsementModel( UserModel user, SkillModel skill) {
```
Comment (Aanchal-Tiwari) [inline skill-tree /src/main/java/com/RDS/skilltree/Endorsement/EndorsementModel.java:39]: [nit]: extra space.

File: skill-tree /src/main/java/com/RDS/skilltree/Endorsement/EndorsementModel.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+import java.time.Instant;
+import java.util.UUID;
+
+@Entity
+@NoArgsConstructor
+@Getter
```
Comment (Aanchal-Tiwari) [inline skill-tree /src/main/java/com/RDS/skilltree/Endorsement/EndorsementModel.java:15]: Explore if data annotation can be used in place of getter and setter.

Comment (DashDeipayan) [reply skill-tree /src/main/java/com/RDS/skilltree/Endorsement/EndorsementModel.java]: changed

**Review by bhtibrewal — COMMENTED** [review]
please add the schema doc link in the description

File: skill-tree /src/main/java/com/RDS/skilltree/User/UserModel.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+    @Column(name = "user_role", nullable = false)
+    private UserRole role;
+
+    @JsonManagedReference
+    @ManyToMany(targetEntity = SkillModel.class, cascade = CascadeType.ALL, fetch = FetchType.EAGER)
+    @JoinTable(name = "user_skill", joinColumns = @JoinColumn(name = "user_id"), inverseJoinColumns = @JoinColumn(name = "skill_id"))
+    private Set<SkillModel> skills;
+
+    public UserModel(String rdsUserId, String firstName, String lastName, URL imageUrl, UserType type, UserRole role) {
```
Comment (vikhyat187) [inline skill-tree /src/main/java/com/RDS/skilltree/User/UserModel.java:48]: @Data annotation creates this, do we need to write it explicitly? 
cc : @DashDeipayan

File: skill-tree /src/main/java/com/RDS/skilltree/Skill/SkillModel.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+    @Column(name = "is_deleted", nullable = false)
+    private boolean deleted;
+
+    @JsonBackReference
+    @JsonIgnore
+    @ManyToMany(mappedBy = "skills", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
+    private Set<UserModel> users;
+
+    public SkillModel(String name, SkillType type, boolean deleted) {
```
Comment (vikhyat187) [inline skill-tree /src/main/java/com/RDS/skilltree/Skill/SkillModel.java:42]: Same as below

File: skill-tree /src/main/java/com/RDS/skilltree/EndorsementList/EndorsementListModel.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    @Column(name = "is_deleted", nullable = false)
+    private boolean isDeleted;
+
+    @Column(name = "type", nullable = false)
+    private EndorsementType type;
+
+    @Column(name = "created_at", nullable = false)
+    private Instant createdAt;
+
+    public EndorsementListModel(EndorsementModel endorsement, UserModel endorser, String description,
```
Comment (vikhyat187) [inline skill-tree /src/main/java/com/RDS/skilltree/EndorsementList/EndorsementListModel.java:42]: same as below

File: skill-tree /src/main/java/com/RDS/skilltree/EndorsementList/EndorsementListModel.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    @Column(name = "description")
+    private String description;
+
+    @Column(name = "is_deleted", nullable = false)
+    private boolean isDeleted;
+
+    @Column(name = "type", nullable = false)
+    private EndorsementType type;
+
+    @Column(name = "created_at", nullable = false)
```
Comment (iamitprakash) [inline skill-tree /src/main/java/com/RDS/skilltree/EndorsementList/EndorsementListModel.java:39]: what about created_by, modified_by/_at

**Review by iamitprakash — COMMENTED** [review]
Please add design doc

File: skill-tree /src/main/java/com/RDS/skilltree/EndorsementList/EndorsementListModel.java (hunk 6 above)
Comment (DashDeipayan) [reply skill-tree /src/main/java/com/RDS/skilltree/EndorsementList/EndorsementListModel.java:39]: Wasn't mentioned in the ER diagram

- https://github.com/Real-Dev-Squad/skill-tree-backend/issues/2

**Review by iamitprakash — APPROVED** [review]

**Review by Aanchal-Tiwari — APPROVED** [review]

**Pull request merged**